src/openms/include/OpenMS/ANALYSIS/ID/IDScoreSwitcherAlgorithm.h

```
@@ -0,0 +1,247 @@
+        String old_score_meta = (old_score_.empty() ? id.getScoreType() :
+                                 old_score_);
+        DataValue dv = hit_it->getMetaValue(old_score_meta);
+        if (!dv.isEmpty()) // meta value for old score already exists
+        {
+          if (fabs((double(dv) - hit_it->getScore()) * 2.0 /
+                   (double(dv) + hit_it->getScore())) > tolerance_)
+          {
+            String msg = "Meta value '" + old_score_meta + "' already exists "
+                                                           "with a conflicting value for " + describeHit_(*hit_it);
+            throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
+                                          msg, dv.toString());
+          } // else: values match, nothing to do
+        }
+        else
+        {
+          hit_it->setMetaValue(old_score_meta, hit_it->getScore());
+        }
+        hit_it->setScore(hit_it->getMetaValue(new_score_));
+      }
+      id.setScoreType(new_score_type_);
+      id.setHigherScoreBetter(higher_better_);
+    }
+
+    /// Looks at the first Hit of the given @p id and according to the given @p type ,
+    /// deduces a fitting score and score direction to be switched to.
+    /// Then tries to switch all hits.
+    void switchToGeneralScoreType(vector<PeptideIdentification>& id, ScoreType type, Size& counter)
+    {
+      if (id.empty()) return;
+      String t = findScoreType(id[0], type);
+      if (t.empty())
+      {
+        String msg = "First encountered ID does not have the requested score type.";
+        throw Exception::MissingInformation(__FILE__, __LINE__,
+                                            OPENMS_PRETTY_FUNCTION, msg);
+      }
+      new_score_ = t;
+      new_score_type_ = t;
+      if (type != ScoreType::RAW && higher_better_ != type_to_better_[type])
+      {
+        OPENMS_LOG_WARN << "Requested non-raw score type does not match the expected score direction. Correcting!\n";
```

does this actually happen? would it be better to fail hard then?

---

I mean, if someone forgets to set the param for the algorithm, this happens. E.g. if you just wanna use this method without setting up the whole algorithm.